tasks/framework-tools/src/frameworkSyncToProject.js

```diff
@@ -1,6 +1,102 @@
-// TODO:
+#!/usr/bin/env node
+/* eslint-env node, es6 */
 
-// Add the deps
-// Run install
-// Copy/ build the files when they change
-// Remove them when user cancels
+/**
+ * - add the deps and install
+ * - build and copy over files
+ * - remove them when user cancels
+ */
+
+const chokidar = require('chokidar')
+const execa = require('execa')
+
+// add the deps and install
+
+const addFrameworkDepsToProject = () => {
+  try {
+    execa.sync('yarn project:deps', {
+      shell: true,
+      stdio: 'inherit',
+      env: {
+        RWJS_CWD: process.env.RWJS_CWD,
+      },
+    })
+  } catch (e) {
+    console.error(
+      'Error: Could not add Redwood Framework dependencies to project'
+    )
+    console.error(e)
+    process.exit(1)
+  }
+}
```

Comment:
I know this is a WIP, but want to give some guidance along the way. I think you can import these from utils and then run the programatically.

Reply:
Guidance is always appreciated! And that was my intention at first, but most of the imperative logic isn't in utils, so I wasn't sure if I should start abstracting some of it into functions or just run things with execa. I started with the latter to get things working consistent and fast, but if we want to be more fine-grained on chokidar's change event (which we do), only the former will let us do that really. So I'm trying to abstract some of the common logic into functions: https://github.com/redwoodjs/redwood/pull/2963/commits/1f4441803395fe9eb1ce6242c45c34184c9d115a#diff-75beb1f310eb89e39495fb6d197c36b122fc5bbce287bfa356129c9f3be3fee3